### PR TITLE
#966: Image support for external image providers/frame validators

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-white/issues/996 |
| Dependencies | https://github.com/ripe-tech/ripe-core/pull/4710 |
| Decisions | - Added support in `image.js` for external image URL providers and frame validators. With this, the thumbnail component just needs to make `bindImage´ with the correct provider and SDK deals with the rest (update parts, change frame, etc)  |
| Animated GIF | Below |

https://user-images.githubusercontent.com/25725586/159502592-c9097962-8f4b-4efd-9388-129398494e2a.mp4

https://user-images.githubusercontent.com/25725586/159502626-bcfff4e5-2d40-4fac-91a9-e8970cf8ccfc.mp4

### CODE REVIEW

File: src/js/base/logic.js [hunk 1]
```diff
@@ -246,6 +246,26 @@ ripe.Ripe.prototype.hasFrame = function(frame) {
...
+ * possible to show that video.
+ * @returns {Boolean} If it's possible to show the video for the
+ * currently loaded model.
+ */
+ripe.Ripe.prototype.hasVideo = function(name) {
+    if (!this.loadedConfig) return true;
+    if (!name) return true;
+
+    return Boolean(this.loadedConfig.videos_m[name]);
+};
```
Comment (joao-conde) [inline src/js/base/logic.js:267]: Maybe:

```js
ripe.Ripe.prototype.hasVideo = function(name) {
    if (!name) return true;
    if (!this.loadedConfig || !this.loadedConfig.videos_m) return true;
    return Boolean(this.loadedConfig.videos_m[name]);
};
```

File: src/js/base/logic.js [hunk 2]
```diff
@@ -246,6 +246,26 @@ ripe.Ripe.prototype.hasFrame = function(frame) {
     return true;
 };
 
+/**
+ * Determines if the model currently loaded (in case there's one) has the
+ * provided video name available in spec.
+ *
+ * Notice that this call does not assure that a video showing is possible
+ * it only determines that according to model's spec it should be
```
Comment (joao-conde) [inline src/js/base/logic.js:254]: "it only determines if according to the model's spec it is possible to get such a video."

**Review by joao-conde — CHANGES_REQUESTED** [review]

File: src/js/base/logic.js (hunk 1 above)
Comment (BeeMargarida) [reply src/js/base/logic.js:267]: I maintained the same structure has the one in `hasFrame`. But I notice that a check is missing, will add

**Review by joao-conde — APPROVED** [review]

**Review by 3rdvision — APPROVED** [review]

**Comment by joamag** [comment]
@BeeMargarida lets add the `bindVideoThumbnail` method that uses `bindImage` to simplify the API ;)

**Review by 3rdvision — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Comment by joamag** [comment]
👏

**Pull request merged**